### DISCUSSION TITLE
The same parameters doesn't give me the same results

### DISCUSSION THREAD

**Discussion by Beewap** [arkypita/LaserGRBL#1301] [Q&A]
So, I'm still new with the laser thing but I don't understand what append.

Few days ago I make some grayscale test and I had a really good result on mdf with this settings :

![grayscale01](https://user-images.githubusercontent.com/77555477/105472088-c9c8d300-5c9b-11eb-92aa-d3abaa59506c.JPG)

And I had this result:
![IMG_20210117_19515222](https://user-images.githubusercontent.com/77555477/105473281-3bede780-5c9d-11eb-8d8c-f0875b49200f.jpg)

Which is pretty good I think.

Now, few days after, I don't know why but I have to modify the setting to get the same result.
From 3000 mm/min (the left one) to 1300mm/min (the right one).

![IMG_20210121_084045 - Copie](https://user-images.githubusercontent.com/77555477/105473627-9c7d2480-5c9d-11eb-914c-ef37819383d1.jpg)

So here is some other information:
I know that I put the laser at 5.5cm. I can move it lower (a bit) or higher and still get the focus point.

Speak of focus point, I don't have a dot, but more a line that I can have really sharp and not blurry. But not a dot. (I can't take a picture of it, the phone only see a dot, but with my eyes I see a really small sharp line).
![IMG_20210122_1040442](https://user-images.githubusercontent.com/77555477/105475862-2c23d280-5ca0-11eb-89b3-0c0281cef538.jpg)

I'm on a win10 computer, Lasergrbl 4.05, firmware 1.1f
I also try with the 4.03 and it is the same issue.
It is a 5.5w laser from Mostics cnc 2 in 1.  (450nm wavelength, 5500mW output I think)

Other information,  I try to engrave another wood type (plywood birch, pin) and there the grayscale value are shorter. 
![IMG_20210121_08522200](https://user-images.githubusercontent.com/77555477/105476934-82454580-5ca1-11eb-927d-b7a66d8d5b92.jpg)

![IMG_20210121_08411221](https://user-images.githubusercontent.com/77555477/105476819-5de96900-5ca1-11eb-9ee3-147237c73da5.jpg)

I don't know what I've done and now that I wanna try some other wood or material, I think that maybe there is something wrong and that's why I got a not so much nice result on birch plywood.

**Comment by arkypita** [comment]
LaserGRBL give the same command to the machine, provided the same configuration.

This is not a LaserGRBL bug or issue, so this should be posted under "[discussion](https://github.com/arkypita/LaserGRBL/discussions)"

Please read https://lasergrbl.com/faq/ where you can find answer to most of this common asked question.

**Comment by Beewap** [comment]
Hi, 
I re-open this post because I still have the issue and didn't find any solution for the moment about the power of my laser.

As I explain before, I had a good results with some parameters when I imported the picture I wanted to engrave and few days later, I had to slow down from 3000mm/min to 1300mm/min to get the same result for the same picture on the same wood without make any change nowhere.

I try to play with some value with $110-111-112 and also with $120-121-122 but it didn't work at all.

Here is my lasergrbl configuration.


$0=10 (Step pulse time)
$1=25 (Step idle delay)
$2=0 (Step pulse invert)
$3=3 (Step direction invert)
$4=0 (Invert step enable pin)
$5=0 (Invert limit pins)
$6=0 (Invert probe pin)
$10=1 (Status report options)
$11=0.010 (Junction deviation)
$12=0.002 (Arc tolerance)
$13=0 (Report in inches)
$20=0 (Soft limits enable)
$21=0 (Hard limits enable)
$22=0 (Homing cycle enable)
$23=0 (Homing direction invert)
$24=25.000 (Homing locate feed rate)
$25=500.000 (Homing search seek rate)
$26=250 (Homing switch debounce delay)
$27=1.000 (Homing switch pull-off distance)
$30=255 (Maximum spindle speed)
$31=0 (Minimum spindle speed)
$32=1 (Laser-mode enable)
$100=800.000 (X-axis travel resolution)
$101=800.000 (Y-axis travel resolution)
$102=800.000 (Z-axis travel resolution)
$110=1000.000 (X-axis maximum rate)
$111=1000.000 (Y-axis maximum rate)
$112=600.000 (Z-axis maximum rate)
$120=30.000 (X-axis acceleration)
$121=30.000 (Y-axis acceleration)
$122=30.000 (Z-axis acceleration)
$130=200.000 (X-axis maximum travel)
$131=200.000 (Y-axis maximum travel)
$132=200.000 (Z-axis maximum travel)

I have already read the faq and I understand the most part I think but it ain't explain why I have to do a so big change on the engraving speed.

Any idea ?

**Comment by arkypita** [comment]
You have set $110/$111 at value of 1000. This means you can ask LaserGRBL to tell the machine to do your job at 1300 as well as 3000, but the machine will only ever do it at maximum speed 1000.

You have set $120/$121=30 (mm/sec^2) this means that your engraver will take several seconds to accelerate up to 1000mm/min speeds.

This combination of parameters will cause your engraver to move much, much slower than set by the Speed value required by LaserGRBL.

This, in combination with the M4 mode (which I imagine you are using) causes the machine to reduce the laser power, as "dynamic power mode" adapts the laser power to the current speed.

It is all written in the faq (section: "My engraver does not reach the speed I set" and "My laser does not reach the power I set") and - I repeat - it is not a problem of LaserGRBL but of how you configured your machine.

**Comment by Beewap** [comment]
Hi,

Thanks for you answer.

I never said that LaserGRBL is the problem, I knew it was something about parameters I've put somehow. I don't remember the settings I had at the first use honestly but the only one I've manually changed is $32 to put 1 and $120 which was on 5000 and I had some blocking issue (I reduce that speed and I never have again this issue).

I have used  cnc files found one the web with some acceleration test, speed test and so .. maybe this settings that are in these files stay in my configuration.

Anyway, I will go to your faq and read the 2 sections you mentionned and try to find my solution.

Thank you :)

ps: I really believe that lasergrbl is an awesome software. Even if I have some issue I really like it. I know that the problem is somewhere between my keyboard and my chair.

**Comment by arkypita** [comment]
In short: you should put higher value in $110/111 and $120/121
Read "[faq](https://lasergrbl.com/faq/)" and "[guide to grbl configuration](http://lasergrbl.com/wp-content/uploads/2020/05/Grbl-Configuration-ENG.pdf)"

**Comment by arkypita** [comment]
While engraving you can see the current speed and power the machine is using.

![image](https://user-images.githubusercontent.com/8782035/105739439-65766f80-5f38-11eb-8db7-8137177e02eb.png)

If you don't see the speed you set, but a lower value, this is a strong indication that your machine configuration is limiting speed.
If the machine configuration limits the speed, the power will also be limited accordingly.

There are no "magic" values to put. Each machine is different from the others, it's up to you to find the most suitable ones for yours. By feeling, I would tell you to start at least with 4000 speed and 800 acceleration (if it's a cnc) but I shot them a bit at random.


Especially CNC with screw movement are very different constructively from belt markers, so they use very different values (CNCs are slower).

Obviously then you cannot expect to make the machine work faster than its physical limits, but now your speed is evidently limited by the configuration.

**Comment by Beewap** [comment]
Thanks.

I'll try to find the good value for this cnc (cnc with screw movement).
Atm looks like more than 4000 in speed and/or more than 6-700 in acceleration is too much. Lots of noises, the head doesn't move as it should and so... With lower value there is no "bad noises".

In any case, if I have good results with 1000 and 200, I guess I can use these speed with no problems.
